### PR TITLE
add spectral unit to moment map 0 display unit, simplify logic

### PR DESCRIPTION
**Pull request by cshanahan1**
There was previously an issue where the moment map plugin was using the spectral y axis unit for moment 0 rather than always using a surface brightness unit. This was fixed in #3156. 

This PR applies some of the changes from #3116, which also fixed that issue but based on an outdated main pre many unit conversion changes, so I found it easier to cherry pick some of the code here rather than rebasing that PR.

The display unit now has the spectral unit because specutils is pinned to the version with the fix for MM units. I made some small tweaks including using _get_display_unit rather than grabbing the UC plugin from the MM plugin and accessing display unit there. 

I think sufficient test coverage was added in #3156 so i didn't add the tests from #3116, but i did expand one of the tests to test that the moment map plugin has the correct units when spectral unit is changed (previously was just testing when flux unit was changed)

This will close JDAT 4651 and #3116 

Closes #3116.

### CODE REVIEW

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:346]: I think this is available from `_get_display_unit` now if you want to make the change here/now

File: jdaviz/core/template_mixin.py
Comment (kecnry) [inline jdaviz/core/template_mixin.py:2956]: is this (and the other removal below) because the marks are handling the unit conversion themselves?

**Review by kecnry — APPROVED** [review]
LGTM - would like to have @bmorris3 review before merging, if possible, just to make sure we're not missing anything from that original PR.

File: jdaviz/core/template_mixin.py
Comment (cshanahan1) [reply jdaviz/core/template_mixin.py:2956]: also see https://jira.stsci.edu/browse/JDAT-4830, this bug was not introduced with this change

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py
Comment (bmorris3) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:1]: Some changes not carried over from @3116:

1. https://github.com/spacetelescope/jdaviz/pull/3116/files#diff-44f616436de8315fb25244c8555ea83596e03bedf23685221f4becc69bad8fbbR32
2. https://github.com/spacetelescope/jdaviz/pull/3116/files#diff-44f616436de8315fb25244c8555ea83596e03bedf23685221f4becc69bad8fbbR112
3. (not sure if this one is still needed?) https://github.com/spacetelescope/jdaviz/pull/3116/files#diff-44f616436de8315fb25244c8555ea83596e03bedf23685221f4becc69bad8fbbR121-R122

**Review by bmorris3 — APPROVED** [review]
Here are some differences with #3116 for @cshanahan1 to double check. Otherwise all good by me.

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py
Comment (cshanahan1) [reply jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:1]: 1. fixed, removed Flux from options
2. i see this correctly removing flux as an option already
3. I intentionally omitted this because I found it to work without it, and I new @kecnry did some work recently on units with an empty viewer so I figured thats why it was now working

**Comment by cshanahan1** [comment]
I have no idea why 3.12 tests are failing

**Comment by pllim** [comment]
I think scipy dev wheel is broken again. You can ignore it.

**Comment by cshanahan1** [comment]
in that case im going to merge this, if i can find some consequence of not subscribing to AddDataMessage i'll follow up but i don't see any issues

**Pull request merged**
